Objects/longobject.c

```diff
@@ -4565,11 +4595,23 @@ _PyLong_Rshift(PyObject *a, size_t shiftby)
 static PyObject *
 long_lshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
 {
-    /* This version due to Tim Peters */
     PyLongObject *z = NULL;
     Py_ssize_t oldsize, newsize, i, j;
     twodigits accum;
 
+    if (wordshift == 0 && IS_MEDIUM_VALUE(a) &&
+        (a->ob_digit[0] & ~(PyLong_MASK >> remshift)) == 0) {
```

Comment:
Why is the `(a->ob_digit[0] & ~(PyLong_MASK >> remshift)) == 0` check necessary?

Reply:
I think it's necessary because `(2**20) << 20` won't fit in a single 30/32-bit digit, and this checks for cases like that.

Reply:
Oh I guess it's not necessary if we use `_PyLong_FromSTwoDigits`

Reply:
It's necessary to ensure the shifting result fits `twodigits` or "medium value" before using `_PyLong_FromSTwoDigits` or something else.

Reply:
`medium_value(a) << remshift` will always fit in an `stwodigits`, though. I think it's safe to drop this condition.

Reply:
Ah, Okay. Thanks.